communication/src/main/java/org/jboss/da/communication/scm/impl/SCMConnectorImpl.java

```diff
@@ -53,4 +54,28 @@
         }
     }
 
+    @Override
+    public Optional<MavenProject> getPom(String scmUrl, String revision, String pomPath)
+            throws ScmException {
+        if (!pomPath.endsWith("pom.xml")) {
+            pomPath += "/pom.xml";
+        }
+
+        try {
+            File tempDir = Files.createTempDirectory("cloned_repo").toFile();
+            try {
+                // git clone
+                // TODO: hardcoded to git right now
+                scmManager.cloneRepository(SCMType.GIT, scmUrl, revision, tempDir.toString());
```

Comment:
We should extract this cloning of repositories to the extra method not to clone the same repository multiple times, but it is a problem in more beans so, we will do it at once. So, it is ok for this PR.